frontend/src/App.tsx

```diff
@@ -1,37 +1,57 @@
 import NavContainer from './components/Layout/NavContainer';
 import Pages from './routes/Pages';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
-import { Fragment } from 'react';
-import { BrowserRouter } from 'react-router-dom';
-import { ToastContainer } from 'react-toastify';
+import { useEffect } from 'react';
+import { useAuth } from 'react-oidc-context';
+import { toast, ToastContainer } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
 
-const queryClient = new QueryClient();
+const useInitDarkMode = () => {
+  const darkMode = localStorage.getItem('darkMode');
+
+  useEffect(() => {
+    if (darkMode === 'true') {
+      document.documentElement.classList.add('dark');
+    } else {
+      localStorage.setItem('darkMode', 'false');
+    }
+  }, [darkMode]);
+};
+
 function App() {
-  if (localStorage.getItem('darkMode') === 'true') {
-    document.documentElement.classList.add('dark');
-  } else {
-    localStorage.setItem('darkMode', 'false');
-  }
+  const auth = useAuth();
+  useInitDarkMode();
+
+  // authentication notifications
+  useEffect(() => {
+    console.log('auth loading: ', auth.isLoading);
+    if (auth.error) {
+      toast(`Oops... ${auth.error.message}`);
+    }
+    switch (auth.activeNavigator) {
+      case 'signinSilent':
+        toast('Signing you in...');
+        break;
+      case 'signoutRedirect':
+        toast(`Signing you out...`);
+    }
+    console.log('authenticated: ', auth.isAuthenticated);
+    if (auth.isAuthenticated) {
+      toast(`Hello ${auth.user?.profile.sub}`);
```

Comment:
This comes quite often and prints a long hexadecimal number in my case. It is also missing translation.